src/Compilers/CSharp/Portable/Compilation/CSharpCompilation.cs

```diff
@@ -2789,6 +2789,37 @@ protected override IPointerTypeSymbol CommonCreatePointerTypeSymbol(ITypeSymbol
             return CreatePointerTypeSymbol(elementType.EnsureCSharpSymbolOrNull<ITypeSymbol, TypeSymbol>("elementType"));
         }
 
+        protected override INamedTypeSymbol CommonCreateTupleTypeSymbol(ITypeSymbol[] elementTypes, string[] elementNames)
```

Comment:
I had looked at other methods in the same class but didn't find a clear standard. But I agree that would be a good thing to have.
Most methods seem to use `IEnumerable`, some use `ImmutableArray`, `ICollection` and a couple use `ArrayBuilder`. I thought some also used arrays, but I was mistaken (they are only `params`).
I'll change to `ImmutableArray`.

I'll also add direct tests for this new API.